great_expectations/cli/cli.py

```diff
@@ -21,50 +17,133 @@
     pass
 
 
-# TODO: consider using a specified-order supporting class for help (but wasn't working with python 2)
-
-
-@click.group()
+class CLIState:
+    def __init__(
+        self,
+        v2_api: bool = False,
+        config_file_location: Optional[str] = None,
+        data_context: DataContext = None,
+    ):
+        self.v2_api = v2_api
+        self.config_file_location = config_file_location
+        self._data_context = data_context
+
+    @property
+    def data_context(self):
+        return self._data_context
+
+    @data_context.setter
+    def data_context(self, data_context):
+        assert isinstance(data_context, DataContext)
+        self._data_context = data_context
+
+    def __repr__(self):
+        return f"CLIState(v2_api={self.v2_api}, config_file_location={self.config_file_location})"
+
+
+class CLI(click.MultiCommand):
+    def list_commands(self, ctx):
+        # note that if --help is called this method is invoked before any flags
+        # are parsed or context set.
+
+        return [
+            "checkpoint",
+            "datasource",
+            "docs",
+            "init",
+            "project",
+            "store",
+            "suite",
+            "validation_operator",
```

Comment:
@Aylr can explain this better than I can, but basically the validation_operator 

> This is AWESOME! Didn't do a code review deep dive - mostly exercised the new Checkpoint functionality for --v3-api and that worked great using the `tmp_cli_uat` repo. Just had a few minor comments, which might be just from my newb understanding of the new CLI architecture changes.
> 
> Regarding "Deprecated suite demo datasource profile", just confirming that this is just for `--v3-api` - got confused for a minute since CLI defaults to `--v2-api` for now.

Correct! `suite demo` and `datasource profile` are only deprecated / removed for `--v3-api` and have not changed for `--v2-api` (which is the default).

Reply:
@Aylr will have more detail here and please correct me if I'm wrong, but basically we still need `validation_operator` here to make sure that the old API can still access it, and we also need a stub `validation_operator.py` file with just `validation_operator = None` in it so it is found when `click` is looking for available commands. As for versioning the help text - I know @Aylr was working on that as well, I think we just haven't found a great solution there yet.